lib/ssh.js

```diff
@@ -3831,8 +3843,20 @@ function parse_KEXINIT(self, callback) {
 function parse_KEX(self, type, callback) {
   var state = self._state,
       instate = state.incoming,
-      payload = instate.payload;
+      outstate = state.outgoing,
+      payload = instate.payload,
+      pktType = (state.kexdh !== 'group') ? DYNAMIC_KEXDH_MESSAGE[type] :
+                                            KEXDH_MESSAGE[type];
 
+  if ((outstate.status === OUT_READY) ||
```

Comment:
Style nit: If complex conditionals need to be split across new lines, the operator should be first on the next line. Also, extra parens should be removed:

```javascript
if (outstate.status === OUT_READY
    || instate.expectedPacket !== pktType) {
```